diff --git a/chrome/browser/ui/views/tabs/common/tab_group_header_view.cc b/chrome/browser/ui/views/tabs/common/tab_group_header_view.cc
index 28fac81220aa6..f160408e8868e 100644
--- a/chrome/browser/ui/views/tabs/common/tab_group_header_view.cc
+++ b/chrome/browser/ui/views/tabs/common/tab_group_header_view.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/tabs/common/tab_group_header_view.h"
 
+#include "base/command_line.h"
 #include "base/task/single_thread_task_runner.h"
 #include "build/build_config.h"
 #include "build/buildflag.h"
@@ -508,7 +509,10 @@ void TabGroupHeaderView::OnDataChanged(
   UpdateIsCollapsed();
   UpdateAccessibleName();
 
-  if (features::IsTabGroupHoverCardsEnabled()) {
+  const bool tab_hover_cards_tooltip =
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          "tab-hover-cards") == "tooltip";
+  if (features::IsTabGroupHoverCardsEnabled() && !tab_hover_cards_tooltip) {
     SetHoverCardDataFrom(tab_group_data);
   } else {
     UpdateTooltipText();
diff --git a/chrome/browser/ui/views/tabs/common/tab_view.cc b/chrome/browser/ui/views/tabs/common/tab_view.cc
index 42c551b09d331..4d38bf1467a79 100644
--- a/chrome/browser/ui/views/tabs/common/tab_view.cc
+++ b/chrome/browser/ui/views/tabs/common/tab_view.cc
@@ -47,6 +47,7 @@
 #include "chrome/browser/ui/views/tabs/tab/alert_indicator_button.h"
 #include "chrome/browser/ui/views/tabs/tab/glow_hover_controller.h"
 #include "chrome/browser/ui/views/tabs/tab/tab_accessibility.h"
+#include "chrome/browser/ui/views/tabs/tab.h"
 #include "chrome/browser/ui/views/tabs/tab/tab_close_button.h"
 #include "chrome/browser/ui/views/tabs/tab/tab_icon.h"
 #include "chrome/browser/ui/views/tabs/tab/tab_title.h"
@@ -596,6 +597,16 @@ void TabView::OnGestureEvent(ui::GestureEvent* event) {
   }
 }
 
+std::u16string TabView::GetRenderedTooltipText(const gfx::Point& p) const {
+  const bool tab_hover_cards_tooltip =
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          "tab-hover-cards") == "tooltip";
+  if (tab_hover_cards_tooltip) {
+    return Tab::GetTooltipText(tab_data_.title, tab_data_.alert_state);
+  }
+  return std::u16string();
+}
+
 void TabView::OnPaint(gfx::Canvas* canvas) {
   // Split pinned tabs have a merged background that is rendered in
   // `SplitTabView`.
diff --git a/chrome/browser/ui/views/tabs/common/tab_view.h b/chrome/browser/ui/views/tabs/common/tab_view.h
index 9a923fe60b3f7..7dcd662f24700 100644
--- a/chrome/browser/ui/views/tabs/common/tab_view.h
+++ b/chrome/browser/ui/views/tabs/common/tab_view.h
@@ -6,6 +6,7 @@
 #define CHROME_BROWSER_UI_VIEWS_TABS_COMMON_TAB_VIEW_H_
 
 #include <optional>
+#include <string>
 #include <vector>
 
 #include "base/callback_list.h"
@@ -24,6 +25,7 @@
 #include "third_party/skia/include/core/SkScalar.h"
 #include "ui/base/metadata/metadata_header_macros.h"
 #include "ui/gfx/canvas.h"
+#include "ui/gfx/geometry/point.h"
 #include "ui/views/context_menu_controller.h"
 #include "ui/views/layout/delegating_layout_manager.h"
 #include "ui/views/layout/flex_layout.h"
@@ -107,6 +109,7 @@ class TabView : public views::View,
   void OnMouseExited(const ui::MouseEvent& event) override;
   bool OnMouseDragged(const ui::MouseEvent& event) override;
   void OnGestureEvent(ui::GestureEvent* event) override;
+  std::u16string GetRenderedTooltipText(const gfx::Point& p) const override;
   void OnPaint(gfx::Canvas* canvas) override;
   void AddedToWidget() override;
   void RemovedFromWidget() override;
diff --git a/chrome/browser/ui/views/tabs/hovercard/tab_hover_card_controller.cc b/chrome/browser/ui/views/tabs/hovercard/tab_hover_card_controller.cc
index 9933d79ae0afd..0acbe4cb0048f 100644
--- a/chrome/browser/ui/views/tabs/hovercard/tab_hover_card_controller.cc
+++ b/chrome/browser/ui/views/tabs/hovercard/tab_hover_card_controller.cc
@@ -305,6 +305,10 @@ bool TabHoverCardController::IsHoverCardShowingForTab(
 void TabHoverCardController::UpdateHoverCard(
     HoverCardAnchorTarget* anchor_target,
     TabSlotController::HoverCardUpdateType update_type) {
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("tab-hover-cards")) {
+    return;
+  }
+
   if (hide_hover_card_lock_count_ > 0) {
     anchor_target = nullptr;
     update_type = TabSlotController::HoverCardUpdateType::kAnimating;
diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index a3eb789db93ef..733f55f44dca8 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -1063,6 +1063,16 @@ void Tab::HideHover(TabStyle::HideHoverStyle style) {
   DeprecatedLayoutImmediately();
 }
 
+std::u16string Tab::GetRenderedTooltipText(const gfx::Point& p) const {
+  const bool tab_hover_cards_tooltip =
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          "tab-hover-cards") == "tooltip";
+  if (tab_hover_cards_tooltip) {
+    return GetTooltipText(data_.title, data_.alert_state);
+  }
+  return std::u16string();
+}
+
 // static
 std::u16string Tab::GetTooltipText(const std::u16string& title,
                                    std::optional<tabs::TabAlert> alert_state) {
diff --git a/chrome/browser/ui/views/tabs/tab.h b/chrome/browser/ui/views/tabs/tab.h
index 4eee7407d9cad..9505948a4b8dc 100644
--- a/chrome/browser/ui/views/tabs/tab.h
+++ b/chrome/browser/ui/views/tabs/tab.h
@@ -108,6 +108,7 @@ class Tab : public gfx::AnimationDelegate,
   void OnMouseEntered(const ui::MouseEvent& event) override;
   void OnMouseExited(const ui::MouseEvent& event) override;
   void OnGestureEvent(ui::GestureEvent* event) override;
+  std::u16string GetRenderedTooltipText(const gfx::Point& p) const override;
   gfx::Size CalculatePreferredSize(
       const views::SizeBounds& available_size) const override;
   void PaintChildren(const views::PaintInfo& info) override;
