diff --git a/chrome/browser/ui/views/tabs/common/tab_view.cc b/chrome/browser/ui/views/tabs/common/tab_view.cc
index 30c8d9158b58e..3d75e66900805 100644
--- a/chrome/browser/ui/views/tabs/common/tab_view.cc
+++ b/chrome/browser/ui/views/tabs/common/tab_view.cc
@@ -7,6 +7,7 @@
 #include <optional>
 #include <string>
 
+#include "base/command_line.h"
 #include "base/functional/callback_helpers.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/metrics/user_metrics.h"
@@ -389,7 +390,10 @@ void TabView::OnMouseReleased(const ui::MouseEvent& event) {
 
   auto* controller = collection_node_->GetController();
   base::WeakPtr<TabView> self = weak_ptr_factory_.GetWeakPtr();
-  if (event.IsOnlyMiddleMouseButton()) {
+  if (event.IsOnlyMiddleMouseButton() ||
+      (event.IsOnlyLeftMouseButton() && event.GetClickCount() == 2 &&
+       base::CommandLine::ForCurrentProcess()->HasSwitch(
+           "double-click-close-tab"))) {
     if (HitTestPoint(event.location())) {
       controller->CloseTab(GetTabInterface());
     }
diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index 8e6aedbbbb635..fb23193e100d9 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -12,6 +12,7 @@
 #include <string_view>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/functional/bind.h"
 #include "base/memory/raw_ptr.h"
 #include "base/metrics/user_metrics.h"
@@ -645,7 +646,10 @@ void Tab::OnMouseReleased(const ui::MouseEvent& event) {
   // Close tab on middle click, but only if the button is released over the tab
   // (normal windows behavior is to discard presses of a UI element where the
   // releases happen off the element).
-  if (event.IsOnlyMiddleMouseButton()) {
+  if (event.IsOnlyMiddleMouseButton() ||
+      (event.IsOnlyLeftMouseButton() && event.GetClickCount() == 2 &&
+       base::CommandLine::ForCurrentProcess()->HasSwitch(
+           "double-click-close-tab"))) {
     if (HitTestPoint(event.location())) {
       controller_->CloseTab(this, CloseTabSource::kFromMouse);
     } else if (closing_) {
