diff --git a/chrome/browser/ui/views/tabs/common/tab_view_horizontal_layout.cc b/chrome/browser/ui/views/tabs/common/tab_view_horizontal_layout.cc
index 0cdc03c73147c..a54f78dc4ca16 100644
--- a/chrome/browser/ui/views/tabs/common/tab_view_horizontal_layout.cc
+++ b/chrome/browser/ui/views/tabs/common/tab_view_horizontal_layout.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/tabs/common/tab_view_horizontal_layout.h"
 
+#include "base/command_line.h"
 #include "chrome/browser/glic/browser_ui/tab_underline_view.h"
 #include "chrome/browser/ui/layout_constants.h"
 #include "chrome/browser/ui/ui_features.h"
@@ -300,6 +301,10 @@ TabViewHorizontalLayout::CalculateChildVisibilities(int width,
   }
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
+  should_show_close_button &=
+      !base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "hide-tab-close-buttons");
+
   if (TabView().active_) {
     if (!declutter_eligible) {
       // Close button is shown on active tabs regardless of the size.
diff --git a/chrome/browser/ui/views/tabs/common/tab_view_vertical_layout.cc b/chrome/browser/ui/views/tabs/common/tab_view_vertical_layout.cc
index 6641560e2f254..9f934b2c9a498 100644
--- a/chrome/browser/ui/views/tabs/common/tab_view_vertical_layout.cc
+++ b/chrome/browser/ui/views/tabs/common/tab_view_vertical_layout.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/tabs/common/tab_view_vertical_layout.h"
 
+#include "base/command_line.h"
 #include "chrome/browser/glic/browser_ui/tab_underline_view.h"
 #include "chrome/browser/ui/layout_constants.h"
 #include "chrome/browser/ui/views/frame/vertical_tab_strip_region_view.h"
@@ -166,6 +167,11 @@ bool TabViewVerticalLayout::IsChildVisible(const views::View* child_view,
   }
 
   if (child_view == TabView().close_button_) {
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons")) {
+      return false;
+    }
+
     if (TabView().pinned_) {
       return false;
     }
diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index d0afedf25815a..c382d26c38252 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -1431,12 +1431,16 @@ void Tab::UpdateIconVisibility() {
 #if BUILDFLAG(IS_CHROMEOS)
   // Hide tab close button for OnTask if locked. Only applicable for non-web
   // browser scenarios.
-  const bool should_show_close_button =
+  bool should_show_close_button =
       !IsLockedForOnTask(controller_->GetBrowserWindowInterface());
 #else
-  const bool should_show_close_button = true;
+  bool should_show_close_button = true;
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
+  should_show_close_button &=
+      !base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "hide-tab-close-buttons");
+
   if (IsActive()) {
     if (!declutter_eligible) {
       // Close button is shown on active tabs regardless of the size.
