diff --git a/chrome/browser/ui/views/tabs/common/tab_view.cc b/chrome/browser/ui/views/tabs/common/tab_view.cc
index a8bd4658c9737..5f1bded41396b 100644
--- a/chrome/browser/ui/views/tabs/common/tab_view.cc
+++ b/chrome/browser/ui/views/tabs/common/tab_view.cc
@@ -854,6 +854,11 @@ bool TabView::IsChildVisible(const views::View* child_view,
       return false;
     }
 
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons")) {
+      return false;
+    }
+
     // When uncollapsing the tabstrip, intentionally start showing the close
     // button on active non-hovered tabs a little bit sooner than reaching the
     // uncollapsed min width, because otherwise the close buttons in a grouped
diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index 83558de4690ec..eb5056f82a4d3 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -1199,6 +1199,10 @@ void Tab::UpdateIconVisibility() {
     // Close button is shown on active tabs regardless of the size.
     showing_close_button_ = true;
 #endif  // BUILDFLAG(IS_CHROMEOS)
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons")) {
+      showing_close_button_ = false;
+    }
 
     if (features::IsTabStripDeclutterEnabled() && showing_close_button_) {
       const bool alert_fits_without_close =
@@ -1257,6 +1261,8 @@ void Tab::UpdateIconVisibility() {
         features::IsTabStripDeclutterEnabled() &&
         available_width <= TabStyle::kTabStripDeclutterMaxTabWidthForCloseHide;
     showing_close_button_ =
+        !base::CommandLine::ForCurrentProcess()->HasSwitch(
+            "hide-tab-close-buttons") &&
 #if BUILDFLAG(IS_CHROMEOS)
         should_show_close_button &&
 #endif
