diff --git a/chrome/browser/extensions/api/developer_private/extension_info_generator.cc b/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
index 64c57f2ca2148..15a942664e68b 100644
--- a/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
+++ b/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
@@ -47,7 +47,6 @@
 #include "extensions/browser/icon_util.h"
 #include "extensions/browser/image_loader.h"
 #include "extensions/browser/managed_installation_mode.h"
-#include "extensions/browser/manifest_v2_handler.h"
 #include "extensions/browser/path_util.h"
 #include "extensions/browser/permissions/site_permissions_helper.h"
 #include "extensions/browser/shared_module_service.h"
@@ -58,7 +57,6 @@
 #include "extensions/common/command.h"
 #include "extensions/common/extension_features.h"
 #include "extensions/common/extension_set.h"
-#include "extensions/common/extension_urls.h"
 #include "extensions/common/install_warning.h"
 #include "extensions/common/manifest.h"
 #include "extensions/common/manifest_handlers/background_info.h"
@@ -872,17 +870,6 @@ void ExtensionInfoGenerator::FillExtensionInfo(const Extension& extension,
         toolbar_actions_model->IsActionPinned(extension.id());
   }
 
-  // MV2 deprecation.
-  ManifestV2Handler* mv2_handler = ManifestV2Handler::Get(profile);
-  CHECK(mv2_handler);
-  info.is_affected_by_mv2_deprecation =
-      mv2_handler->IsExtensionAffected(extension);
-  if (info.web_store_url.length() > 0) {
-    info.recommendations_url =
-        extension_urls::GetNewWebstoreItemRecommendationsUrl(extension.id())
-            .spec();
-  }
-
   info.can_upload_as_account_extension =
       AccountExtensionTracker::Get(profile)->CanUploadAsAccountExtension(
           extension)
diff --git a/chrome/browser/ui/webui/extensions/extensions_ui.cc b/chrome/browser/ui/webui/extensions/extensions_ui.cc
index 0a0eb5d94e971..38e4db931580f 100644
--- a/chrome/browser/ui/webui/extensions/extensions_ui.cc
+++ b/chrome/browser/ui/webui/extensions/extensions_ui.cc
@@ -488,9 +488,7 @@ content::WebUIDataSource* CreateAndAddExtensionsSource(Profile* profile,
       "enableUserPermittedSites",
       base::FeatureList::IsEnabled(
           extensions_features::kExtensionsMenuAccessControlWithPermittedSites));
-  source->AddBoolean(
-      "safetyHubThreeDotDetails",
-      base::FeatureList::IsEnabled(features::kSafetyHubThreeDotDetails));
+  source->AddBoolean("safetyHubThreeDotDetails", false);
   source->AddBoolean("enableGlobalScopedShortcuts", IsGlobalShortcutEnabled());
 
   // MV2 deprecation.
diff --git a/extensions/browser/manifest_v2_handler.cc b/extensions/browser/manifest_v2_handler.cc
index 5841d3dfbc888..e0fb4e06b97bc 100644
--- a/extensions/browser/manifest_v2_handler.cc
+++ b/extensions/browser/manifest_v2_handler.cc
@@ -109,12 +109,7 @@ bool ManifestV2HandlerFactory::ServiceIsCreatedWithBrowserContext() const {
 
 // Returns true if legacy extensions should be disabled.
 bool ShouldDisableLegacyExtensions() {
-  if (g_allow_mv2_for_testing) {
-    // We allow legacy MV2 extensions for testing purposes.
-    return false;
-  }
-
-  return true;
+  return false;
 }
 
 }  // namespace
diff --git a/extensions/common/extension.cc b/extensions/common/extension.cc
index 9413abec978e1..060eb46371fa5 100644
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -97,12 +97,6 @@ bool IsManifestSupported(int manifest_version,
   // Supported versions are always safe.
   if (manifest_version >= kMinimumSupportedManifestVersion &&
       manifest_version <= kMaximumSupportedManifestVersion) {
-    // Emit a warning for unpacked extensions on Manifest V2 warning that
-    // MV2 is deprecated.
-    if (type == Manifest::Type::kExtension && manifest_version == 2 &&
-        Manifest::IsUnpackedLocation(location)) {
-      *warning = errors::kManifestV2IsDeprecatedWarning;
-    }
     return true;
   }
 
