diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index 438d0fa748339..575c5a25f82a6 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -389,7 +389,6 @@ if (!is_android && !is_mac) {
     data_deps += [
       "//chrome/browser/resources/media/mei_preload:component",
       "//chrome/browser/web_applications/isolated_web_apps/key_distribution/preload:component",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component",
       "//third_party/widevine/cdm",
     ]
 
@@ -1389,7 +1388,6 @@ if (is_win) {
       "//chrome/browser/lens/sapisid:sapisid_bundle_data",
       "//chrome/browser/resources/media/mei_preload:component_bundle",
       "//chrome/browser/web_applications/isolated_web_apps/key_distribution/preload:component_bundle",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component_bundle",
     ]
 
     if (is_chrome_branded) {
diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index eb192f7220e5e..4010528bc88d9 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -1840,7 +1840,6 @@ if (_is_default_toolchain) {
   java_group("chrome_public_non_pak_assets") {
     deps = [
       "//chrome/android/webapk/libs/runtime_library:runtime_library_assets",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:privacy_sandbox_attestations_assets",
     ]
   }
 
diff --git a/chrome/browser/component_updater/registration.cc b/chrome/browser/component_updater/registration.cc
index 9fd11c9f722cb..b7926ade35a67 100644
--- a/chrome/browser/component_updater/registration.cc
+++ b/chrome/browser/component_updater/registration.cc
@@ -28,7 +28,6 @@
 #include "chrome/browser/component_updater/mei_preload_component_installer.h"
 #include "chrome/browser/component_updater/pki_metadata_component_installer.h"
 #include "chrome/browser/component_updater/platform_runtime_component_installer.h"
-#include "chrome/browser/component_updater/privacy_sandbox_attestations_component_installer.h"
 #include "chrome/browser/component_updater/private_verification_tokens_installer.h"
 #include "chrome/browser/component_updater/ssl_error_assistant_component_installer.h"
 #include "chrome/browser/component_updater/subresource_filter_component_installer.h"
@@ -154,7 +153,6 @@ void RegisterComponentsForUpdate() {
   RegisterTrustTokenKeyCommitmentsComponentIfTrustTokensEnabled(cus);
   RegisterPrivateVerificationTokensComponentIfEnabled(cus);
   RegisterFirstPartySetsComponent(cus);
-  RegisterPrivacySandboxAttestationsComponent(cus);
   RegisterActorSafetyListsComponent(cus, base::OnceClosure());
   if (history_embeddings::IsHistoryEmbeddingsFeatureEnabled()) {
     RegisterHistorySearchStringsComponent(cus);
diff --git a/chrome/browser/resources/BUILD.gn b/chrome/browser/resources/BUILD.gn
index 494f08cb1820a..0067a6e0cc3e2 100644
--- a/chrome/browser/resources/BUILD.gn
+++ b/chrome/browser/resources/BUILD.gn
@@ -135,7 +135,6 @@ group("resources") {
     public_deps += [
       "unexportable_keys_internals:resources",
       "//chrome/browser/resources/media/mei_preload:component",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component",
     ]
   }
 
diff --git a/chrome/installer/linux/BUILD.gn b/chrome/installer/linux/BUILD.gn
index 47a27ce391364..4a46d18bce92b 100644
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -157,8 +157,6 @@ packaging_files = packaging_files_binaries + [
                     "$root_out_dir/locales/en-US.pak",
                     "$root_out_dir/MEIPreload/manifest.json",
                     "$root_out_dir/MEIPreload/preloaded_data.pb",
-                    "$root_out_dir/PrivacySandboxAttestationsPreloaded/manifest.json",
-                    "$root_out_dir/PrivacySandboxAttestationsPreloaded/privacy-sandbox-attestations.dat",
                   ]
 if (enable_swiftshader) {
   packaging_files += [ "$root_out_dir/vk_swiftshader_icd.json" ]
@@ -530,7 +528,6 @@ group("installer_deps") {
     "//chrome/browser/enterprise/connectors/device_trust/key_management/installer/management_service:chrome_management_service",
     "//chrome/browser/resources/media/mei_preload:component",
     "//components/crash/core/app:chrome_crashpad_handler",
-    "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component",
     "//sandbox/linux:chrome_sandbox",
     "//third_party/devtools-frontend/src/inspector_overlay:build_inspector_overlay",
     "//ui/resources:copy_ui_resources_100_percent",
diff --git a/chrome/installer/linux/common/installer.py b/chrome/installer/linux/common/installer.py
index 1710e45870945..cc321032d967b 100644
--- a/chrome/installer/linux/common/installer.py
+++ b/chrome/installer/linux/common/installer.py
@@ -586,24 +586,6 @@ class Installer:
         for pak in (self.output_dir / "locales").glob("*.pak"):
             self._install_into_dir(pak, locales_dir, mode=0o644)
 
-        # Privacy Sandbox Attestation
-        psa_manifest = (
-            self.output_dir
-            / "PrivacySandboxAttestationsPreloaded/manifest.json"
-        )
-        if psa_manifest.exists():
-            psa_dir = install_dir / "PrivacySandboxAttestationsPreloaded"
-            psa_dir.mkdir(parents=True, exist_ok=True)
-            psa_dir.chmod(0o755)
-            self._install_into_dir(psa_manifest, psa_dir, mode=0o644)
-            self._install_into_dir(
-                self.output_dir
-                / "PrivacySandboxAttestationsPreloaded"
-                / "privacy-sandbox-attestations.dat",
-                psa_dir,
-                mode=0o644,
-            )
-
         # MEI Preload
         mei_manifest = self.output_dir / "MEIPreload/manifest.json"
         if mei_manifest.exists():
diff --git a/chrome/installer/mini_installer/chrome.release b/chrome/installer/mini_installer/chrome.release
index 1811767f803c3..bc2c2ab9b54f0 100644
--- a/chrome/installer/mini_installer/chrome.release
+++ b/chrome/installer/mini_installer/chrome.release
@@ -88,12 +88,6 @@ WidevineCdm\_platform_specific\win_arm64\widevinecdm.dll.sig: %(VersionDir)s\Wid
 Logo.png: %(VersionDir)s\VisualElements\
 SmallLogo.png: %(VersionDir)s\VisualElements\
 
-#
-# Privacy Sandbox Attestation
-#
-PrivacySandboxAttestationsPreloaded\manifest.json: %(VersionDir)s\PrivacySandboxAttestationsPreloaded\
-PrivacySandboxAttestationsPreloaded\privacy-sandbox-attestations.dat: %(VersionDir)s\PrivacySandboxAttestationsPreloaded\
-
 #
 # MEI Preload sub dir
 #
diff --git a/components/privacy_sandbox/privacy_sandbox_features.cc b/components/privacy_sandbox/privacy_sandbox_features.cc
index 11ff98e135767..04eaa94b10729 100644
--- a/components/privacy_sandbox/privacy_sandbox_features.cc
+++ b/components/privacy_sandbox/privacy_sandbox_features.cc
@@ -16,7 +16,7 @@ BASE_FEATURE(kDefaultAllowPrivacySandboxAttestations,
 
 #if BUILDFLAG(IS_ANDROID)
 BASE_FEATURE(kPrivacySandboxAttestationsLoadFromAPKAsset,
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 #endif  // BUILDFLAG(IS_ANDROID)
 
 BASE_FEATURE(kRelatedWebsiteSetsUi, base::FEATURE_DISABLED_BY_DEFAULT);
diff --git a/content/common/features.cc b/content/common/features.cc
index 7edb9b3eb69a8..839f6103d26ff 100644
--- a/content/common/features.cc
+++ b/content/common/features.cc
@@ -584,7 +584,7 @@ BASE_FEATURE(kPriorityOverridePendingViews, base::FEATURE_ENABLED_BY_DEFAULT);
 // Enables exposure of the core milestone 1 (M1) APIs in the renderer without an
 // origin trial token: Attribution Reporting, FLEDGE, Topics.
 BASE_FEATURE(kPrivacySandboxAdsAPIsM1Override,
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 #if BUILDFLAG(IS_ANDROID)
 // When disabled("legacy behavior") it resets ongoing gestures when window loses
diff --git a/third_party/blink/common/features.cc b/third_party/blink/common/features.cc
index a25fcdff1c6f6..10b907c13e662 100644
--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -122,7 +122,7 @@ BASE_FEATURE(kAllowDatapipeDrainedAsBytesConsumerInBFCache,
 
 // Enables URN URLs like those produced by Protected Audience auctions to be
 // displayed by iframes (instead of requiring fenced frames).
-BASE_FEATURE(kAllowURNsInIframes, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kAllowURNsInIframes, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // A console warning is shown when the opaque url returned from Protected
 // Audience/selectUrl is used to navigate an iframe. Since fenced frames are not
