diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index f02c32aa3dec3..8e0a1072c00c0 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1399,7 +1399,6 @@ if (is_win) {
       ":widevine_cdm_library",
       "//chrome/browser/resources/media/mei_preload:component_bundle",
       "//chrome/browser/web_applications/isolated_web_apps/key_distribution/preload:component_bundle",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component_bundle",
     ]
 
     if (is_chrome_branded) {
diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index 92faf6c6420d7..14fd6a17f2cd0 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -1845,7 +1845,6 @@ if (_is_default_toolchain) {
   java_group("chrome_public_non_pak_assets") {
     deps = [
       "//chrome/android/webapk/libs/runtime_library:runtime_library_assets",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:privacy_sandbox_attestations_assets",
     ]
   }
 
diff --git a/chrome/browser/component_updater/registration.cc b/chrome/browser/component_updater/registration.cc
index 9cf5a226c7f74..395f9abe18e47 100644
--- a/chrome/browser/component_updater/registration.cc
+++ b/chrome/browser/component_updater/registration.cc
@@ -27,7 +27,6 @@
 #include "chrome/browser/component_updater/mei_preload_component_installer.h"
 #include "chrome/browser/component_updater/pki_metadata_component_installer.h"
 #include "chrome/browser/component_updater/platform_runtime_component_installer.h"
-#include "chrome/browser/component_updater/privacy_sandbox_attestations_component_installer.h"
 #include "chrome/browser/component_updater/private_verification_tokens_installer.h"
 #include "chrome/browser/component_updater/ssl_error_assistant_component_installer.h"
 #include "chrome/browser/component_updater/subresource_filter_component_installer.h"
@@ -159,7 +158,6 @@ void RegisterComponentsForUpdate() {
   RegisterTrustTokenKeyCommitmentsComponentIfTrustTokensEnabled(cus);
   RegisterPrivateVerificationTokensComponentIfEnabled(cus);
   RegisterFirstPartySetsComponent(cus);
-  RegisterPrivacySandboxAttestationsComponent(cus);
   RegisterActorSafetyListsComponent(cus, base::OnceClosure());
   if (history_embeddings::IsHistoryEmbeddingsFeatureEnabled()) {
     RegisterHistorySearchStringsComponent(cus);
diff --git a/components/privacy_sandbox/privacy_sandbox_features.cc b/components/privacy_sandbox/privacy_sandbox_features.cc
index 11ff98e135767..04eaa94b10729 100644
--- a/components/privacy_sandbox/privacy_sandbox_features.cc
+++ b/components/privacy_sandbox/privacy_sandbox_features.cc
@@ -16,7 +16,7 @@ BASE_FEATURE(kDefaultAllowPrivacySandboxAttestations,
 
 #if BUILDFLAG(IS_ANDROID)
 BASE_FEATURE(kPrivacySandboxAttestationsLoadFromAPKAsset,
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 #endif  // BUILDFLAG(IS_ANDROID)
 
 BASE_FEATURE(kRelatedWebsiteSetsUi, base::FEATURE_DISABLED_BY_DEFAULT);
diff --git a/content/common/features.cc b/content/common/features.cc
index 27f7314ba2503..236d7702da53f 100644
--- a/content/common/features.cc
+++ b/content/common/features.cc
@@ -604,7 +604,7 @@ BASE_FEATURE(kPriorityOverridePendingViews, base::FEATURE_ENABLED_BY_DEFAULT);
 // Enables exposure of the core milestone 1 (M1) APIs in the renderer without an
 // origin trial token: Attribution Reporting, FLEDGE, Topics.
 BASE_FEATURE(kPrivacySandboxAdsAPIsM1Override,
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 #if BUILDFLAG(IS_ANDROID)
 // When disabled("legacy behavior") it resets ongoing gestures when window loses
diff --git a/third_party/blink/common/features.cc b/third_party/blink/common/features.cc
index 248061c9a678d..6aa5d49158ea8 100644
--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -129,7 +129,7 @@ BASE_FEATURE(kAllowDatapipeDrainedAsBytesConsumerInBFCache,
 
 // Enables URN URLs like those produced by Protected Audience auctions to be
 // displayed by iframes (instead of requiring fenced frames).
-BASE_FEATURE(kAllowURNsInIframes, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kAllowURNsInIframes, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // A console warning is shown when the opaque url returned from Protected
 // Audience/selectUrl is used to navigate an iframe. Since fenced frames are not
