diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index 438d0fa748339..575c5a25f82a6 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -389,7 +389,6 @@ if (!is_android && !is_mac) {
     data_deps += [
       "//chrome/browser/resources/media/mei_preload:component",
       "//chrome/browser/web_applications/isolated_web_apps/key_distribution/preload:component",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component",
       "//third_party/widevine/cdm",
     ]
 
@@ -1389,7 +1388,6 @@ if (is_win) {
       "//chrome/browser/lens/sapisid:sapisid_bundle_data",
       "//chrome/browser/resources/media/mei_preload:component_bundle",
       "//chrome/browser/web_applications/isolated_web_apps/key_distribution/preload:component_bundle",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component_bundle",
     ]
 
     if (is_chrome_branded) {
diff --git a/chrome/browser/resources/BUILD.gn b/chrome/browser/resources/BUILD.gn
index 494f08cb1820a..0067a6e0cc3e2 100644
--- a/chrome/browser/resources/BUILD.gn
+++ b/chrome/browser/resources/BUILD.gn
@@ -135,7 +135,6 @@ group("resources") {
     public_deps += [
       "unexportable_keys_internals:resources",
       "//chrome/browser/resources/media/mei_preload:component",
-      "//components/privacy_sandbox/privacy_sandbox_attestations/preload:component",
     ]
   }
 
