diff --git a/components/dom_distiller/core/css/distilledpage_common.css b/components/dom_distiller/core/css/distilledpage_common.css
index 570369d1febfe..f80502ac392de 100644
--- a/components/dom_distiller/core/css/distilledpage_common.css
+++ b/components/dom_distiller/core/css/distilledpage_common.css
@@ -1,4 +1,4 @@
-/* Copyright 2014 The Chromium Authors
+/* Copyright 2026 The Chromium Authors and Alex313031
  * Use of this source code is governed by a BSD-style license that can be
  * found in the LICENSE file. */
 
@@ -26,6 +26,12 @@
   box-sizing: border-box;
 }
 
+/* Material Icons rendering fix */
+.material-icons {
+  font-family: Material Icons;
+  font-style: normal !important;
+}
+
 /* Remove all margins and padding from certain element and add word wrap. */
 
 blockquote,
@@ -282,7 +288,7 @@ video::-webkit-media-controls-fullscreen-button {
 /* Iframe sizing. */
 .youtubeContainer {
   height: 0;
-  /* This is the perecnt height of a standard HD video. */
+  /* This is the percent height of a standard HD video. */
   padding-bottom: 56.25%;
   position: relative;
   width: 100%;
diff --git a/components/dom_distiller/core/dom_distiller_features.cc b/components/dom_distiller/core/dom_distiller_features.cc
index 26e59179487a3..8df76d8163b67 100644
--- a/components/dom_distiller/core/dom_distiller_features.cc
+++ b/components/dom_distiller/core/dom_distiller_features.cc
@@ -1,4 +1,4 @@
-// Copyright 2015 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -21,13 +21,19 @@
 namespace dom_distiller {
 
 bool IsDomDistillerEnabled() {
+  static const bool kReaderModeDesktop =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reader-mode");
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableDomDistiller);
+             switches::kEnableDomDistiller) ||
+         kReaderModeDesktop;
 }
 
 bool ShouldStartDistillabilityService() {
+  static const bool kReaderModeDesktop =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reader-mode");
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
-      switches::kEnableDistillabilityService);
+             switches::kEnableDistillabilityService) ||
+         kReaderModeDesktop;
 }
 
 
