diff --git a/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_utils.cc b/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_utils.cc
index f7c81500b2e9f..ea6463178d243 100644
--- a/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_utils.cc
+++ b/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_utils.cc
@@ -1,4 +1,4 @@
-// Copyright 2021 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -27,7 +27,7 @@
 #endif
 
 namespace {
-bool force_activation_for_testing = false;
+bool force_activation_for_testing = true;
 }
 
 bool IsFeatureSupportedOnChannel(const LabInfo& lab) {
@@ -140,8 +140,14 @@ bool AreNewChromeLabsExperimentsAvailable(Profile* profile) {
 }
 
 bool IsChromeLabsEnabled() {
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "disable-chrome-labs")) {
+    return false;
+  }
+
   // Always early out on the stable channel regardless of other conditions.
-  if (chrome::GetChannel() == version_info::Channel::STABLE) {
+  if (chrome::GetChannel() == version_info::Channel::STABLE &&
+      !force_activation_for_testing) {
     return false;
   }
 #if !BUILDFLAG(IS_ANDROID)
@@ -166,7 +172,7 @@ bool IsChromeLabsEnabled() {
   }
 
   // The percentage of users that should see the feature.
-  const int kChromeLabsActivationPercentage = 99;
+  constexpr int kChromeLabsActivationPercentage = 100;
 
   if (force_activation_for_testing ||
       g_browser_process->local_state()->GetInteger(
