diff --git a/chrome/browser/ui/views/frame/browser_view.cc b/chrome/browser/ui/views/frame/browser_view.cc
index a24bb10309206..332216c05ce71 100644
--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -1339,8 +1339,11 @@ bool BrowserView::GetTabStripVisible() const {
 }
 
 bool BrowserView::ShouldDrawTabStrokes() const {
+  static const bool force_disable_tab_outlines =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "force-disable-tab-outlines");
 #if BUILDFLAG(IS_CHROMEOS)
-  return false;
+  return !force_disable_tab_outlines;
 #else   // BUILDFLAG(IS_CHROMEOS)
 
   if (web_app::AppBrowserController::From(browser()) &&
@@ -1362,7 +1365,7 @@ bool BrowserView::ShouldDrawTabStrokes() const {
         theme_service->UsingSystemTheme();
   }
 
-  if (!using_system_theme) {
+  if (!using_system_theme && force_disable_tab_outlines) {
     // Strokes are only supported for system themes.
     return false;
   }
@@ -1380,7 +1383,8 @@ bool BrowserView::ShouldDrawTabStrokes() const {
       GetFrameView()->GetFrameColor(BrowserFrameActiveState::kActive);
   const float contrast_ratio =
       color_utils::GetContrastRatio(background_color, frame_color);
-  return contrast_ratio < kMinimumContrastRatioForOutlines;
+  return contrast_ratio < kMinimumContrastRatioForOutlines ||
+         !force_disable_tab_outlines;
 #endif  // !BUILDFLAG(IS_CHROMEOS)
 }
 
