diff --git a/chrome/browser/renderer_context_menu/render_view_context_menu.cc b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
index 35fb31bca4217..2f97e8dd118e9 100644
--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -2321,7 +2321,9 @@ void RenderViewContextMenu::AppendVideoItems() {
     menu_model_.AddItemWithStringId(IDC_CONTENT_CONTEXT_PICTUREINPICTURE,
                                     IDS_CONTENT_CONTEXT_PICTUREINPICTURE);
 
-    if (media_router::MediaRouterEnabled(browser_context_)) {
+    if (media_router::MediaRouterEnabled(browser_context_) &&
+        !base::FeatureList::IsEnabled(
+            features::kThoriumHideCastContextMenu)) {
       menu_model_.AddItemWithStringIdAndIcon(
           IDC_ROUTE_MEDIA, IDS_MEDIA_ROUTER_MENU_ITEM_TITLE,
           ui::ImageModel::FromVectorIcon(kCastChromeRefreshOldIcon,
@@ -2733,7 +2735,9 @@ void RenderViewContextMenu::AppendTranslateItem() {
 }
 
 void RenderViewContextMenu::AppendMediaRouterItem() {
-  if (media_router::MediaRouterEnabled(browser_context_)) {
+  if (media_router::MediaRouterEnabled(browser_context_) &&
+      !base::FeatureList::IsEnabled(
+          features::kThoriumHideCastContextMenu)) {
     AddItemWithOptionalIcon(IDC_ROUTE_MEDIA, IDS_MEDIA_ROUTER_MENU_ITEM_TITLE,
                             kCastChromeRefreshOldIcon);
   }
diff --git a/chrome/browser/ui/ui_features.cc b/chrome/browser/ui/ui_features.cc
index 81b550e796c3b..6eef560442151 100644
--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -213,6 +213,10 @@ BASE_FEATURE_PARAM(std::string,
 BASE_FEATURE(kThoriumDisableQRCodeGenerator,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+// Hides Cast entries from the rendered-page context menu.
+BASE_FEATURE(kThoriumHideCastContextMenu,
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
 #if BUILDFLAG(IS_WIN)
 // Enables the UI for Process Isolation in chrome://settings/system.
 BASE_FEATURE(kProcessIsolationSettings, base::FEATURE_DISABLED_BY_DEFAULT);
diff --git a/chrome/browser/ui/ui_features.h b/chrome/browser/ui/ui_features.h
index 1a0bfd381827a..423c0a44b8cac 100644
--- a/chrome/browser/ui/ui_features.h
+++ b/chrome/browser/ui/ui_features.h
@@ -402,6 +402,9 @@ BASE_DECLARE_FEATURE_PARAM(std::string, kOpenOmniboxUrlInNewTabMode);
 // Disables the QR code generator entry points for sharing page links.
 BASE_DECLARE_FEATURE(kThoriumDisableQRCodeGenerator);
 
+// Hides Cast entries from the rendered-page context menu.
+BASE_DECLARE_FEATURE(kThoriumHideCastContextMenu);
+
 BASE_DECLARE_FEATURE(kVerticalTabsGrabHandleRemoval);
 BASE_DECLARE_FEATURE_PARAM(bool, kVerticalTabsGrabHandleRemovalAlways);
 
