diff --git a/chrome/browser/ui/bookmarks/bookmark_utils_desktop.cc b/chrome/browser/ui/bookmarks/bookmark_utils_desktop.cc
index ba74c2ef1a948..0a62f3b8dfb55 100644
--- a/chrome/browser/ui/bookmarks/bookmark_utils_desktop.cc
+++ b/chrome/browser/ui/bookmarks/bookmark_utils_desktop.cc
@@ -66,7 +66,7 @@ using bookmarks::BookmarkNode;
 
 namespace bookmarks {
 
-size_t kNumBookmarkUrlsBeforePrompting = 15;
+size_t kNumBookmarkUrlsBeforePrompting = 50;
 
 static BookmarkNavigationWrapper* g_nav_wrapper_test_instance = nullptr;
 
