diff --git a/chrome/app/chrome_command_ids.h b/chrome/app/chrome_command_ids.h
index 9b9707ce49c38..d6cd825b1d498 100644
--- a/chrome/app/chrome_command_ids.h
+++ b/chrome/app/chrome_command_ids.h
@@ -326,6 +326,7 @@
 #define IDC_CONTENT_CONTEXT_OPENLINKINPROFILE 50108
 #define IDC_CONTENT_CONTEXT_OPENLINKBOOKMARKAPP 50109
 #define IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW 50111
+#define IDC_CONTENT_CONTEXT_OPENLINKSAMETAB 50119
 #define IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST 50112
 // Image items.
 #define IDC_CONTENT_CONTEXT_SAVEIMAGEAS 50120
diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index 8ee24b5e26fc2..b0a1d84286ff9 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -722,6 +722,9 @@ are declared in tools/grit/grit_args.gni.
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWTAB" desc="The name of the Open Link in New Tab command in the content area context menu">
             Open link in new &amp;tab
           </message>
+          <message name="IDS_CONTENT_CONTEXT_OPENLINKSAMETAB" desc="The name of the Open Link in Current Tab command in the content area context menu">
+            Open link in &amp;current tab
+          </message>
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWWINDOW" desc="The name of the Open Link in New Window command in the content area context menu">
             Open link in new &amp;window
           </message>
@@ -1066,6 +1069,9 @@ are declared in tools/grit/grit_args.gni.
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWTAB" desc="In Title Case: The name of the Open Link in New Tab command in the content area context menu">
             Open Link in New &amp;Tab
           </message>
+          <message name="IDS_CONTENT_CONTEXT_OPENLINKSAMETAB" desc="In Title Case: The name of the Open Link in Current Tab command in the content area context menu">
+            Open Link in &amp;Current Tab
+          </message>
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWWINDOW" desc="In Title Case: The name of the Open Link in New Window command in the content area context menu">
             Open Link in New &amp;Window
           </message>
diff --git a/chrome/browser/renderer_context_menu/render_view_context_menu.cc b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
index ce8f38b2fbdfb..82a68d268f85f 100644
--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -642,13 +642,14 @@ const std::map<int, int>& GetIdcToUmaMap(UmaEnumIdLookupType type) {
        {IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST, 166},
        {IDC_SPELLCHECK_REMOVE_FROM_DICTIONARY, 167},
        {IDC_CONTENT_CONTEXT_SAVE_TO_MEMORY_BANKS, 168},
+       {IDC_CONTENT_CONTEXT_OPENLINKSAMETAB, 169},
        // To add new items:
        //   - Add one more line above this comment block, using the UMA value
        //     from the line below this comment block.
        //   - Increment the UMA value in that latter line.
        //   - Add the new item to the RenderViewContextMenuItem enum in
        //     tools/metrics/histograms/metadata/ui/enums.xml.
-       {0, 169}});
+       {0, 170}});
   // LINT.ThenChange(//tools/metrics/histograms/metadata/ui/enums.xml:RenderViewContextMenuItem)
 
   // LINT.IfChange(ContextMenuOptionDesktop)
@@ -688,13 +689,14 @@ const std::map<int, int>& GetIdcToUmaMap(UmaEnumIdLookupType type) {
        {IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW, 31},
        {IDC_CONTENT_CONTEXT_GLICSHAREIMAGE, 32},
        {IDC_SPELLCHECK_REMOVE_FROM_DICTIONARY, 33},
+       {IDC_CONTENT_CONTEXT_OPENLINKSAMETAB, 34},
        // To add new items:
        //   - Add one more line above this comment block, using the UMA value
        //     from the line below this comment block.
        //   - Increment the UMA value in that latter line.
        //   - Add the new item to the ContextMenuOptionDesktop enum in
        //     tools/metrics/histograms/metadata/ui/enums.xml.
-       {0, 33}});
+       {0, 34}});
   // LINT.ThenChange(//tools/metrics/histograms/metadata/ui/enums.xml:ContextMenuOptionDesktop)
 
   return *(type == UmaEnumIdLookupType::GeneralEnumId ? kGeneralMap
@@ -756,7 +758,8 @@ int FindUMAEnumValueForCommand(int id, UmaEnumIdLookupType type) {
 
 // Returns true if the command id is for opening a link.
 bool IsCommandForOpenLink(int id) {
-  return id == IDC_CONTENT_CONTEXT_OPENLINKNEWTAB ||
+  return id == IDC_CONTENT_CONTEXT_OPENLINKSAMETAB ||
+         id == IDC_CONTENT_CONTEXT_OPENLINKNEWTAB ||
          id == IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW ||
          id == IDC_CONTENT_CONTEXT_OPENLINKOFFTHERECORD ||
          id == IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW ||
@@ -1979,7 +1982,12 @@ void RenderViewContextMenu::AppendLinkItems() {
       }
     }
 #endif
-
+    if (show_open_in_new_tab) {
+      menu_model_.AddItemWithStringId(
+          IDC_CONTENT_CONTEXT_OPENLINKSAMETAB,
+          in_app ? IDS_CONTENT_CONTEXT_OPENLINKNEWTAB_INAPP
+                 : IDS_CONTENT_CONTEXT_OPENLINKSAMETAB);
+    }
     if (params_.link_url.is_valid()) {
       AppendProtocolHandlerSubMenu();
     }
@@ -3390,6 +3398,7 @@ bool RenderViewContextMenu::IsCommandIdEnabled(int id) const {
 
     case IDC_CONTENT_CONTEXT_OPENLINKBOOKMARKAPP:
     case IDC_CONTENT_CONTEXT_OPENLINKINPROFILE:
+    case IDC_CONTENT_CONTEXT_OPENLINKSAMETAB:
     case IDC_CONTENT_CONTEXT_OPENLINKNEWTAB:
     case IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW:
       return navigation_allowed && params_.link_url.is_valid() &&
@@ -3714,6 +3723,30 @@ void RenderViewContextMenu::ExecuteCommand(int id, int event_flags) {
   }
 
   switch (id) {
+    case IDC_CONTENT_CONTEXT_OPENLINKSAMETAB: {
+      WindowOpenDisposition new_tab_disposition =
+          WindowOpenDisposition::CURRENT_TAB;
+      BrowserWindowInterface* browser = nullptr;
+      if (IsInProgressiveWebApp()) {
+        browser = FindNormalBrowser(GetProfile());
+        new_tab_disposition = browser
+                                  ? WindowOpenDisposition::NEW_FOREGROUND_TAB
+                                  : WindowOpenDisposition::NEW_WINDOW;
+      }
+
+      OpenURLParams params = GetOpenURLParamsWithExtraHeaders(
+          params_.link_url, params_.frame_url, params_.frame_origin,
+          new_tab_disposition, ui::PAGE_TRANSITION_LINK,
+          /*extra_headers=*/std::string(), /*started_from_context_menu=*/true);
+
+      if (browser) {
+        browser->OpenURL(params, /*navigation_handle_callback=*/{});
+      } else {
+        source_web_contents_->OpenURL(params,
+                                      /*navigation_handle_callback=*/{});
+      }
+      break;
+    }
     case IDC_CONTENT_CONTEXT_OPENLINKNEWTAB: {
       WindowOpenDisposition new_tab_disposition =
           WindowOpenDisposition::NEW_BACKGROUND_TAB;
diff --git a/chrome/browser/ui/actions/chrome_action_id.h b/chrome/browser/ui/actions/chrome_action_id.h
index f6b247d5a60a6..4568dbcb034cb 100644
--- a/chrome/browser/ui/actions/chrome_action_id.h
+++ b/chrome/browser/ui/actions/chrome_action_id.h
@@ -220,6 +220,7 @@
   E(kActionWritingDirectionRtl, IDC_WRITING_DIRECTION_RTL) \
   E(kActionHideApp, IDC_HIDE_APP) \
   /* Link items. */ \
+  E(kActionContentContextOpenLinkSameTab, IDC_CONTENT_CONTEXT_OPENLINKSAMETAB) \
   E(kActionContentContextOpenLinkNewTab, IDC_CONTENT_CONTEXT_OPENLINKNEWTAB) \
   E(kActionContentContextOpenLinkNewWindow, \
     IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW) \
diff --git a/tools/metrics/histograms/metadata/ui/enums.xml b/tools/metrics/histograms/metadata/ui/enums.xml
index 3d9f7a81534a6..7c471f14acb5b 100644
--- a/tools/metrics/histograms/metadata/ui/enums.xml
+++ b/tools/metrics/histograms/metadata/ui/enums.xml
@@ -79,6 +79,7 @@ chromium-metrics-reviews@google.com.
   <int value="31" label="Open link in split view"/>
   <int value="32" label="Create image with Gemini in Chrome"/>
   <int value="33" label="Remove from dictionary"/>
+  <int value="34" label="Open link in this tab"/>
 </enum>
 
 <!-- LINT.ThenChange(//chrome/browser/renderer_context_menu/render_view_context_menu.cc:ContextMenuOptionDesktop) -->
@@ -376,6 +377,7 @@ chromium-metrics-reviews@google.com.
   <int value="166" label="IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST"/>
   <int value="167" label="IDC_SPELLCHECK_REMOVE_FROM_DICTIONARY"/>
   <int value="168" label="IDC_CONTENT_CONTEXT_SAVE_TO_MEMORY_BANKS"/>
+  <int value="169" label="IDC_CONTENT_CONTEXT_OPENLINKSAMETAB"/>
 </enum>
 
 <!-- LINT.ThenChange(//chrome/browser/renderer_context_menu/render_view_context_menu.cc:RenderViewContextMenuItem) -->
