diff --git a/chrome/app/chrome_command_ids.h b/chrome/app/chrome_command_ids.h
index 0e9128249b48c..03eecd37d16ac 100644
--- a/chrome/app/chrome_command_ids.h
+++ b/chrome/app/chrome_command_ids.h
@@ -328,6 +328,7 @@
 #define IDC_CONTENT_CONTEXT_OPENLINKINPROFILE 50108
 #define IDC_CONTENT_CONTEXT_OPENLINKBOOKMARKAPP 50109
 #define IDC_CONTENT_CONTEXT_OPENLINKSPLITVIEW 50111
+#define IDC_CONTENT_CONTEXT_OPENLINKSAMETAB 50119
 #define IDC_CONTENT_CONTEXT_ADD_LINK_TO_READING_LIST 50112
 #define IDC_CONTENT_CONTEXT_OPENLINK_ISOLATED 50113
 // Image items.
diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index a70a1e7fe80d0..1a1d9b744f19c 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -811,6 +811,9 @@ are declared in tools/grit/grit_args.gni.
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWTAB" desc="The name of the Open Link in New Tab command in the content area context menu">
             Open link in new &amp;tab
           </message>
+          <message name="IDS_CONTENT_CONTEXT_OPENLINKSAMETAB" desc="The name of the Open Link in Current Tab command in the content area context menu">
+            Open link in &amp;current tab
+          </message>
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWWINDOW" desc="The name of the Open Link in New Window command in the content area context menu">
             Open link in new &amp;window
           </message>
@@ -1167,6 +1170,9 @@ are declared in tools/grit/grit_args.gni.
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWTAB" desc="In Title Case: The name of the Open Link in New Tab command in the content area context menu">
             Open Link in New &amp;Tab
           </message>
+          <message name="IDS_CONTENT_CONTEXT_OPENLINKSAMETAB" desc="In Title Case: The name of the Open Link in Current Tab command in the content area context menu">
+            Open Link in &amp;Current Tab
+          </message>
           <message name="IDS_CONTENT_CONTEXT_OPENLINKNEWWINDOW" desc="In Title Case: The name of the Open Link in New Window command in the content area context menu">
             Open Link in New &amp;Window
           </message>
diff --git a/chrome/browser/renderer_context_menu/render_view_context_menu.cc b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
index b6c3e9dacf530..5c3417585e495 100644
--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -656,13 +656,14 @@ int UmaEnumForCommand(int key, UmaEnumIdLookupType type) {
        {IDC_SPELLCHECK_REMOVE_FROM_DICTIONARY, 167},
        {IDC_CONTENT_CONTEXT_SAVE_TO_MEMORY_BANKS, 168},
        {IDC_CONTENT_CONTEXT_OPENLINK_ISOLATED, 169},
+       {IDC_CONTENT_CONTEXT_OPENLINKSAMETAB, 170},
        // To add new items:
        //   - Add one more line above this comment block, using the UMA value
        //     from the line below this comment block.
        //   - Increment the UMA value in that latter line.
        //   - Add the new item to the RenderViewContextMenuItem enum in
        //     tools/metrics/histograms/metadata/ui/enums.xml.
-       {kUmaMaxValueKey, 170}});
+       {kUmaMaxValueKey, 171}});
   // LINT.ThenChange(//tools/metrics/histograms/metadata/ui/enums.xml:RenderViewContextMenuItem)
 
   // LINT.IfChange(ContextMenuOptionDesktop)
@@ -703,13 +704,14 @@ int UmaEnumForCommand(int key, UmaEnumIdLookupType type) {
        {IDC_CONTENT_CONTEXT_GLICSHAREIMAGE, 32},
        {IDC_SPELLCHECK_REMOVE_FROM_DICTIONARY, 33},
        {IDC_CONTENT_CONTEXT_OPENLINK_ISOLATED, 34},
+       {IDC_CONTENT_CONTEXT_OPENLINKSAMETAB, 35},
        // To add new items:
        //   - Add one more line above this comment block, using the UMA value
        //     from the line below this comment block.
        //   - Increment the UMA value in that latter line.
        //   - Add the new item to the ContextMenuOptionDesktop enum in
        //     tools/metrics/histograms/metadata/ui/enums.xml.
-       {kUmaMaxValueKey, 35}});
+       {kUmaMaxValueKey, 36}});
   // LINT.ThenChange(//tools/metrics/histograms/metadata/ui/enums.xml:ContextMenuOptionDesktop)
 
   switch (type) {
@@ -773,7 +775,8 @@ int FindUMAEnumValueForCommand(int id, UmaEnumIdLookupType type) {
 
 // Returns true if the command id is for opening a link.
 bool IsCommandForOpenLink(int id) {
-  return id == IDC_CONTENT_CONTEXT_OPENLINKNEWTAB ||
+  return id == IDC_CONTENT_CONTEXT_OPENLINKSAMETAB ||
+         id == IDC_CONTENT_CONTEXT_OPENLINKNEWTAB ||
          id == IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW ||
          id == IDC_CONTENT_CONTEXT_OPENLINKOFFTHERECORD ||
          id == IDC_CONTENT_CONTEXT_OPENLINK_ISOLATED ||
@@ -2024,7 +2027,12 @@ void RenderViewContextMenu::AppendLinkItems() {
       }
     }
 #endif
-
+    if (show_open_in_new_tab) {
+      menu_model_.AddItemWithStringId(
+          IDC_CONTENT_CONTEXT_OPENLINKSAMETAB,
+          in_app ? IDS_CONTENT_CONTEXT_OPENLINKNEWTAB_INAPP
+                 : IDS_CONTENT_CONTEXT_OPENLINKSAMETAB);
+    }
     if (params_.link_url.is_valid()) {
       AppendProtocolHandlerSubMenu();
     }
@@ -3454,6 +3462,7 @@ bool RenderViewContextMenu::IsCommandIdEnabled(int id) const {
 
     case IDC_CONTENT_CONTEXT_OPENLINKBOOKMARKAPP:
     case IDC_CONTENT_CONTEXT_OPENLINKINPROFILE:
+    case IDC_CONTENT_CONTEXT_OPENLINKSAMETAB:
     case IDC_CONTENT_CONTEXT_OPENLINKNEWTAB:
     case IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW:
       return navigation_allowed && params_.link_url.is_valid() &&
@@ -3784,6 +3793,30 @@ void RenderViewContextMenu::ExecuteCommand(int id, int event_flags) {
   }
 
   switch (id) {
+    case IDC_CONTENT_CONTEXT_OPENLINKSAMETAB: {
+      WindowOpenDisposition new_tab_disposition =
+          WindowOpenDisposition::CURRENT_TAB;
+      BrowserWindowInterface* browser = nullptr;
+      if (IsInProgressiveWebApp()) {
+        browser = FindNormalBrowser(GetProfile());
+        new_tab_disposition = browser
+                                  ? WindowOpenDisposition::NEW_FOREGROUND_TAB
+                                  : WindowOpenDisposition::NEW_WINDOW;
+      }
+
+      OpenURLParams params = GetOpenURLParamsWithExtraHeaders(
+          params_.link_url, params_.frame_url, params_.frame_origin,
+          new_tab_disposition, ui::PAGE_TRANSITION_LINK,
+          /*extra_headers=*/std::string(), /*started_from_context_menu=*/true);
+
+      if (browser) {
+        browser->OpenURL(params, /*navigation_handle_callback=*/{});
+      } else {
+        source_web_contents_->OpenURL(params,
+                                      /*navigation_handle_callback=*/{});
+      }
+      break;
+    }
     case IDC_CONTENT_CONTEXT_OPENLINKNEWTAB: {
       WindowOpenDisposition new_tab_disposition =
           WindowOpenDisposition::NEW_BACKGROUND_TAB;
diff --git a/chrome/browser/ui/actions/chrome_action_id.h b/chrome/browser/ui/actions/chrome_action_id.h
index 954c2933aaf30..f9dd35ac99e18 100644
--- a/chrome/browser/ui/actions/chrome_action_id.h
+++ b/chrome/browser/ui/actions/chrome_action_id.h
@@ -225,6 +225,7 @@
   E(kActionWritingDirectionRtl, IDC_WRITING_DIRECTION_RTL) \
   E(kActionHideApp, IDC_HIDE_APP) \
   /* Link items. */ \
+  E(kActionContentContextOpenLinkSameTab, IDC_CONTENT_CONTEXT_OPENLINKSAMETAB) \
   E(kActionContentContextOpenLinkNewTab, IDC_CONTENT_CONTEXT_OPENLINKNEWTAB) \
   E(kActionContentContextOpenLinkNewWindow, \
     IDC_CONTENT_CONTEXT_OPENLINKNEWWINDOW) \
diff --git a/tools/metrics/histograms/metadata/ui/enums.xml b/tools/metrics/histograms/metadata/ui/enums.xml
index 193a907199770..22a7402335fea 100644
--- a/tools/metrics/histograms/metadata/ui/enums.xml
+++ b/tools/metrics/histograms/metadata/ui/enums.xml
@@ -80,6 +80,7 @@ chromium-metrics-reviews@google.com.
   <int value="32" label="Create image with Gemini in Chrome"/>
   <int value="33" label="Remove from dictionary"/>
   <int value="34" label="Open link in isolated window"/>
+  <int value="35" label="Open link in this tab"/>
 </enum>
 
 <!-- LINT.ThenChange(//chrome/browser/renderer_context_menu/render_view_context_menu.cc:ContextMenuOptionDesktop) -->
@@ -414,6 +415,7 @@ chromium-metrics-reviews@google.com.
   <int value="167" label="IDC_SPELLCHECK_REMOVE_FROM_DICTIONARY"/>
   <int value="168" label="IDC_CONTENT_CONTEXT_SAVE_TO_MEMORY_BANKS"/>
   <int value="169" label="IDC_CONTENT_CONTEXT_OPENLINK_ISOLATED"/>
+  <int value="170" label="IDC_CONTENT_CONTEXT_OPENLINKSAMETAB"/>
 </enum>
 
 <!-- LINT.ThenChange(//chrome/browser/renderer_context_menu/render_view_context_menu.cc:RenderViewContextMenuItem) -->
