diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index de060ada24258..33c9dbac3c165 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -363,6 +363,7 @@ Tab::Tab(tabs::TabHandle handle, TabSlotController* controller)
   SetID(VIEW_ID_TAB);
 
   AddChildViewRaw(title_.get());
+  title_->SetFontList(tab_style_views_->GetFontList());
 
   SetEventTargeter(std::make_unique<views::ViewTargeter>(this));
 
@@ -1069,6 +1070,7 @@ void Tab::ActiveStateChanged() {
   icon_->SetActiveState(IsActive());
   alert_indicator_button_->OnParentTabButtonColorChanged();
   alert_indicator_button_->UpdateEnabledForMuteToggle();
+  title_->SetFontList(tab_style_views_->GetFontList());
   DeprecatedLayoutImmediately();
 }
 
diff --git a/chrome/browser/ui/views/tabs/tab_style_views.cc b/chrome/browser/ui/views/tabs/tab_style_views.cc
index 9ed4e9a31b776..31b171c3cac2d 100644
--- a/chrome/browser/ui/views/tabs/tab_style_views.cc
+++ b/chrome/browser/ui/views/tabs/tab_style_views.cc
@@ -9,6 +9,7 @@
 #include <string>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/strings/strcat.h"
 #include "base/strings/string_util.h"
 #include "base/strings/utf_string_conversions.h"
@@ -16,6 +17,8 @@
 #include "chrome/browser/ui/views/tabs/common/horizontal_tab_style_views.h"
 #include "chrome/browser/ui/views/tabs/common/vertical_tab_style_views.h"
 #include "ui/gfx/color_utils.h"
+#include "ui/views/style/typography.h"
+#include "ui/views/style/typography_provider.h"
 
 // static
 std::u16string ui::metadata::TypeConverter<TabStyle::TabColors>::ToString(
@@ -74,3 +77,14 @@ std::unique_ptr<TabStyleViews> TabStyleViews::Create(
 TabStyleViews::TabStyleViews() : tab_style_(TabStyle::Get()) {}
 
 TabStyleViews::~TabStyleViews() = default;
+
+const gfx::FontList& TabStyleViews::GetFontList() const {
+  static const bool prominent_active_tab_titles =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "prominent-active-tab-titles");
+  const int context = prominent_active_tab_titles && delegate()->IsActive()
+                          ? views::style::CONTEXT_BUTTON_MD
+                          : views::style::CONTEXT_LABEL;
+  return views::TypographyProvider::Get().GetFont(context,
+                                                 views::style::STYLE_PRIMARY);
+}
diff --git a/chrome/browser/ui/views/tabs/tab_style_views.h b/chrome/browser/ui/views/tabs/tab_style_views.h
index b3cd296d9073e..ccfae977801f1 100644
--- a/chrome/browser/ui/views/tabs/tab_style_views.h
+++ b/chrome/browser/ui/views/tabs/tab_style_views.h
@@ -84,6 +84,7 @@ class TabStyleViewDelegate {
 };
 
 namespace gfx {
+class FontList;
 class Canvas;
 }
 
@@ -117,6 +118,9 @@ class TabStyleViews {
   virtual std::optional<SkPath> GetChildClipPath(
       float paint_recording_scale) const = 0;
 
+  // Returns the appropriate font for the current active state.
+  const gfx::FontList& GetFontList() const;
+
   // Paints the tab.
   virtual void PaintTab(gfx::Canvas* canvas) const = 0;
 
