diff --git a/chrome/browser/ui/views/tabs/tab.cc b/chrome/browser/ui/views/tabs/tab.cc
index 0d15d3e76dbcd..c90349d854592 100644
--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -234,6 +234,7 @@ Tab::Tab(tabs::TabHandle handle, TabSlotController* controller)
   SetID(VIEW_ID_TAB);
 
   AddChildViewRaw(title_.get());
+  title_->SetFontList(tab_style_views_->GetFontList());
 
   SetEventTargeter(std::make_unique<views::ViewTargeter>(this));
 
@@ -917,6 +918,7 @@ void Tab::ActiveStateChanged() {
   icon_->SetActiveState(IsActive());
   alert_indicator_button_->OnParentTabButtonColorChanged();
   alert_indicator_button_->UpdateEnabledForMuteToggle();
+  title_->SetFontList(tab_style_views_->GetFontList());
   DeprecatedLayoutImmediately();
 }
 
diff --git a/chrome/browser/ui/views/tabs/tab_style_views.cc b/chrome/browser/ui/views/tabs/tab_style_views.cc
index e65c106f7e634..8282c1ca20111 100644
--- a/chrome/browser/ui/views/tabs/tab_style_views.cc
+++ b/chrome/browser/ui/views/tabs/tab_style_views.cc
@@ -9,6 +9,7 @@
 #include <utility>
 
 #include "base/check_is_test.h"
+#include "base/command_line.h"
 #include "base/i18n/rtl.h"
 #include "base/memory/raw_ptr.h"
 #include "base/numerics/safe_conversions.h"
@@ -42,6 +43,7 @@
 #include "ui/gfx/geometry/insets.h"
 #include "ui/gfx/geometry/skia_conversions.h"
 #include "ui/gfx/scoped_canvas.h"
+#include "ui/views/style/typography.h"
 #include "ui/views/widget/widget.h"
 
 namespace {
@@ -66,6 +68,7 @@ class TabStyleViewsImpl : public TabStyleViews {
   bool IsApparentlyActive() const override;
   float GetCurrentActiveOpacity() const override;
   TabStyle::TabColors CalculateTargetColors() const override;
+  const gfx::FontList& GetFontList() const override;
   void ShowHover(TabStyle::ShowHoverStyle style) override;
   void HideHover(TabStyle::HideHoverStyle style) override;
 
@@ -164,6 +167,9 @@ class TabStyleViewsImpl : public TabStyleViews {
   const raw_ptr<const Tab> tab_;
 
   std::unique_ptr<GlowHoverController> hover_controller_;
+
+  gfx::FontList normal_font_;
+  gfx::FontList heavy_font_;
 };
 
 // TabStyleViewsImpl ----------------------------------------------------------
@@ -172,7 +178,11 @@ TabStyleViewsImpl::TabStyleViewsImpl(Tab* tab)
     : tab_(tab),
       hover_controller_((tab && gfx::Animation::ShouldRenderRichAnimation())
                             ? new GlowHoverController(tab)
-                            : nullptr) {
+                            : nullptr),
+      normal_font_(views::TypographyProvider::Get().GetFont(
+          views::style::CONTEXT_LABEL, views::style::STYLE_PRIMARY)),
+      heavy_font_(views::TypographyProvider::Get().GetFont(
+          views::style::CONTEXT_BUTTON_MD, views::style::STYLE_PRIMARY)) {
   // `tab_` must not be nullptr.
   CHECK(tab_);
   // TODO(dfried): create a new STYLE_PROMINENT or similar to use instead of
@@ -599,6 +609,18 @@ TabStyle::TabColors TabStyleViewsImpl::CalculateTargetColors() const {
       tab()->GetColorProvider());
 }
 
+const gfx::FontList& TabStyleViewsImpl::GetFontList() const {
+  static const bool prominent_active_tab_titles =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "prominent-active-tab-titles");
+
+  if (prominent_active_tab_titles && tab_->IsActive()) {
+    return heavy_font_;
+  }
+
+  return normal_font_;
+}
+
 void TabStyleViewsImpl::ShowHover(TabStyle::ShowHoverStyle style) {
   if (!hover_controller_) {
     return;
diff --git a/chrome/browser/ui/views/tabs/tab_style_views.h b/chrome/browser/ui/views/tabs/tab_style_views.h
index 2a29c948f9869..87b038336d86e 100644
--- a/chrome/browser/ui/views/tabs/tab_style_views.h
+++ b/chrome/browser/ui/views/tabs/tab_style_views.h
@@ -52,6 +52,9 @@ class TabStyleViews {
                          float scale,
                          const TabPathFlags& flags) const = 0;
 
+  // Returns the appropriate font for the current active state.
+  virtual const gfx::FontList& GetFontList() const = 0;
+
   // Paints the tab.
   virtual void PaintTab(gfx::Canvas* canvas) const = 0;
 
