diff --git a/components/embedder_support/user_agent_utils.cc b/components/embedder_support/user_agent_utils.cc
index fcf9cc329a799..475a79b01e6f9 100644
--- a/components/embedder_support/user_agent_utils.cc
+++ b/components/embedder_support/user_agent_utils.cc
@@ -438,6 +438,8 @@ std::string GetOSVersion(IncludeAndroidBuildNumber include_android_build_number,
 #elif BUILDFLAG(IS_ANDROID)
                       "%s%s", android_version_str.c_str(),
                       android_info_str.c_str()
+#elif BUILDFLAG(IS_LINUX) && defined(ARCH_CPU_ARM64)
+                      "13597.84.0"
 #else
                       ""
 #endif
@@ -830,6 +832,11 @@ std::string BuildOSCpuInfoFromOSVersionAndCpuType(const std::string& os_version,
 #elif BUILDFLAG(IS_IOS)
                       "CPU %s OS %s like Mac OS X", cpu_type.c_str(),
                       os_version.c_str()
+#elif BUILDFLAG(IS_LINUX) && defined(ARCH_CPU_ARM64)
+                      "CrOS "
+                      "%s %s",
+                      cpu_type.c_str(),
+                      os_version.c_str()
 #elif BUILDFLAG(IS_POSIX)
                       "%s %s",
                       unixinfo.sysname,  // e.g. Linux
