diff --git a/chrome/browser/ui/tabs/tab_style.cc b/chrome/browser/ui/tabs/tab_style.cc
index 6f58e50c84858..19f8af9e512b7 100644
--- a/chrome/browser/ui/tabs/tab_style.cc
+++ b/chrome/browser/ui/tabs/tab_style.cc
@@ -122,10 +122,20 @@ int TabStyle::GetMinimumInactiveWidth() const {
 }
 
 int TabStyle::GetTopCornerRadius() const {
+  static const bool rectangular_tabs =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("rectangular-tabs");
+  if (rectangular_tabs) {
+    return 4;
+  }
   return features::IsThorium2024() ? 8 : 10;
 }
 
 int TabStyle::GetBottomCornerRadius() const {
+  static const bool rectangular_tabs =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("rectangular-tabs");
+  if (rectangular_tabs) {
+    return 4;
+  }
   return features::IsThorium2024() ? 8 : 12;
 }
 
diff --git a/chrome/browser/ui/views/tabs/shared/tab_strip_combo_button.cc b/chrome/browser/ui/views/tabs/shared/tab_strip_combo_button.cc
index ee64772b32f63..f279127701458 100644
--- a/chrome/browser/ui/views/tabs/shared/tab_strip_combo_button.cc
+++ b/chrome/browser/ui/views/tabs/shared/tab_strip_combo_button.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/tabs/shared/tab_strip_combo_button.h"
 
+#include "base/command_line.h"
 #include "base/i18n/rtl.h"
 #include "base/metrics/user_metrics.h"
 #include "base/metrics/user_metrics_action.h"
@@ -107,6 +108,10 @@ TabStripComboButton::TabStripComboButton(BrowserWindowInterface* browser,
     end_button_->SetUseThorium2024Style(true);
     end_button_->SetIconSize(kThorium2024TabSearchIconSize);
   }
+  if (context_ == Context::kHorizontalTabStrip &&
+      base::CommandLine::ForCurrentProcess()->HasSwitch("rectangular-tabs")) {
+    end_button_->SetUseRectangularStyle(true);
+  }
 
   PrefService* prefs = browser_->GetProfile()->GetPrefs();
   pref_registrar_.Init(prefs);
diff --git a/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.cc b/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.cc
index be89deb33c207..7c240ecd7b40b 100644
--- a/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.cc
+++ b/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.cc
@@ -148,6 +148,20 @@ void TabStripFlatEdgeButton::SetUseThorium2024Style(
   SchedulePaint();
 }
 
+void TabStripFlatEdgeButton::SetUseRectangularStyle(
+    bool use_rectangular_style) {
+  if (use_rectangular_style_ == use_rectangular_style) {
+    return;
+  }
+  use_rectangular_style_ = use_rectangular_style;
+  SetProperty(views::kHighlightPathGeneratorKey,
+              std::make_unique<views::RoundRectHighlightPathGenerator>(
+                  gfx::Insets(), GetButtonCornerRadii()));
+  views::InkDrop::Get(this)->GetInkDrop()->HostSizeChanged(size());
+
+  SchedulePaint();
+}
+
 void TabStripFlatEdgeButton::SetExpansionFactor(float factor) {
   if (expansion_factor_ == factor) {
     return;
@@ -300,6 +314,10 @@ ui::ColorId TabStripFlatEdgeButton::GetBackgroundColor() const {
 }
 
 gfx::RoundedCornersF TabStripFlatEdgeButton::GetButtonCornerRadii() const {
+  if (use_rectangular_style_) {
+    return gfx::RoundedCornersF(4.0f);
+  }
+
   const float kFlatRadius = use_thorium_2024_style_ ? 4.0f : 2.0f;
   const float kRoundedRadius = use_thorium_2024_style_ ? 14.0f : 10.0f;
   float flat_radius = kFlatRadius + ((kRoundedRadius - kFlatRadius) *
diff --git a/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.h b/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.h
index 1065282429cbd..00b2247ad054b 100644
--- a/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.h
+++ b/chrome/browser/ui/views/tabs/shared/tab_strip_flat_edge_button.h
@@ -42,6 +42,7 @@ class TabStripFlatEdgeButton : public views::LabelButton,
   void SetInsets(const gfx::Insets& insets);
   void SetIconOpacity(float opacity);
   void SetUseThorium2024Style(bool use_thorium_2024_style);
+  void SetUseRectangularStyle(bool use_rectangular_style);
 
   void SetExpansionFactor(float factor);
   float GetExpansionFactor() const { return expansion_factor_; }
@@ -89,6 +90,7 @@ class TabStripFlatEdgeButton : public views::LabelButton,
   FlatEdge flat_edge_ = FlatEdge::kNone;
   base::CallbackListSubscription paint_as_active_subscription_;
   bool use_thorium_2024_style_ = false;
+  bool use_rectangular_style_ = false;
 
   base::RepeatingClosureList will_invoke_action_callback_list_;
 };
diff --git a/chrome/browser/ui/views/tabs/tab_strip_control_button.cc b/chrome/browser/ui/views/tabs/tab_strip_control_button.cc
index 5fe159795bab6..0047c8532da8a 100644
--- a/chrome/browser/ui/views/tabs/tab_strip_control_button.cc
+++ b/chrome/browser/ui/views/tabs/tab_strip_control_button.cc
@@ -7,6 +7,7 @@
 #include <optional>
 #include <utility>
 
+#include "base/command_line.h"
 #include "chrome/browser/ui/browser_window/public/browser_window_interface.h"
 #include "chrome/browser/ui/views/frame/browser_frame_view.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
@@ -193,7 +194,9 @@ void TabStripControlButton::UpdateIcon() {
 }
 
 int TabStripControlButton::GetCornerRadius() const {
-  return TabStripControlButton::kButtonSize.width() / 2;
+  static const bool rectangular_tabs =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("rectangular-tabs");
+  return rectangular_tabs ? 4 : TabStripControlButton::kButtonSize.width() / 2;
 }
 
 int TabStripControlButton::GetFlatCornerRadius() const {
