diff --git a/components/omnibox/browser/BUILD.gn b/components/omnibox/browser/BUILD.gn
index 66b93310a4820..dc55566bd9561 100644
--- a/components/omnibox/browser/BUILD.gn
+++ b/components/omnibox/browser/BUILD.gn
@@ -114,6 +114,13 @@ aggregate_vector_icons("omnibox_vector_icons") {
     "wb_sunny_filled.icon",
   ]
 
+  if (is_thorium_build) {
+    sources += [
+      "find_in_page_thorium.icon",
+      "http_chrome_refresh_thorium.icon",
+    ]
+  }
+
   if (is_mac) {
     sources += [
       "ios_share.icon",
diff --git a/components/vector_icons/BUILD.gn b/components/vector_icons/BUILD.gn
index 33d25f74926d9..4b93fb1087626 100644
--- a/components/vector_icons/BUILD.gn
+++ b/components/vector_icons/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2017 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -522,6 +522,24 @@ if (current_toolchain == default_toolchain) {
       "work_old.icon",
     ]
 
+    if (is_thorium_build) {
+      sources += [
+        "bookmark_folder.icon",
+        "folder_managed.icon",
+        "info_regular.icon",
+        "thorium/close_chrome_refresh_thorium.icon",
+        "thorium/cookie_chrome_refresh_thorium.icon",
+        "thorium/extension_chrome_refresh_thorium.icon",
+        "thorium/back_arrow_chrome_refresh_thorium.icon",
+        "thorium/forward_arrow_chrome_refresh_thorium.icon",
+        "thorium/reload_chrome_refresh_thorium.icon",
+        "thorium/restore_tab.icon",
+        "thorium/submenu_arrow_chrome_refresh_thorium.icon",
+        "thorium/volume_off_thorium.icon",
+        "thorium/volume_off_chrome_refresh_thorium.icon",
+        "thorium/volume_up_chrome_refresh_thorium.icon",
+      ]
+    }
     if (is_chromeos) {
       sources += [ "notification_download_old.icon" ]
     }
