diff --git a/chrome/app/vector_icons/BUILD.gn b/chrome/app/vector_icons/BUILD.gn
index af65e7bc8601f..4dc688f8fe9fd 100644
--- a/chrome/app/vector_icons/BUILD.gn
+++ b/chrome/app/vector_icons/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2017 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -366,6 +366,7 @@ aggregate_vector_icons("chrome_vector_icons") {
     "translate_old.icon",
     "trash_can_old.icon",
     "trash_can_refresh_old.icon",
+    "trash_can_regular.icon",
     "travel_explore.icon",
     "travel_explore_old.icon",
     "tv.icon",
@@ -420,6 +421,19 @@ aggregate_vector_icons("chrome_vector_icons") {
     # go/keep-sorted end
   ]
 
+  if (is_thorium_build) {
+    sources += [
+      "thorium/browser_tools_chrome_refresh_thorium.icon",
+      "thorium/incognito_thorium.icon",
+      "thorium/incognito_refresh_menu_thorium.icon",
+      "thorium/media_toolbar_button_chrome_refresh_thorium.icon",
+      "thorium/media_toolbar_button_touch_thorium.icon",
+      "thorium/navigate_home_chrome_refresh_thorium.icon",
+      "thorium/navigate_stop_chrome_refresh_thorium.icon",
+      "thorium/science_thorium.icon",
+      "thorium/trash_can_refresh_thorium.icon",
+    ]
+  }
   if (is_mac) {
     sources += [
       "add_circle_filled.icon",
