diff --git a/third_party/blink/common/features.cc b/third_party/blink/common/features.cc
index 37ad7b41c296a..304d8e6942fef 100644
--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -1,4 +1,4 @@
-// Copyright 2018 The Chromium Authors
+// Copyright 2026 The Chromium Authors and Alex313031
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -106,7 +106,8 @@ BASE_FEATURE_ENUM_PARAM(
 // Exports this feature so that Rust code can use it.
 extern "C" {
 // Block all MIDI access with the MIDI_SYSEX permission
-BASE_FEATURE(kBlockMidiByDefault, base::FEATURE_ENABLED_BY_DEFAULT);
+// Allow MIDI for Thorium Legacy
+BASE_FEATURE(kBlockMidiByDefault, base::FEATURE_DISABLED_BY_DEFAULT);
 }
 
 BASE_FEATURE(kComputePressureRateObfuscationMitigation,
@@ -687,7 +688,7 @@ BASE_FEATURE(kFetchDestinationJsonCssModules,
              "kFetchDestinationJsonCssModules",
              base::FEATURE_ENABLED_BY_DEFAULT);
 
-BASE_FEATURE(kFileSystemUrlNavigation, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kFileSystemUrlNavigation, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kFilteringScrollPrediction,
 #if BUILDFLAG(IS_ANDROID)
@@ -2196,7 +2197,7 @@ BASE_FEATURE(
 #if BUILDFLAG(IS_ANDROID)
     // TODO(crbug.com/351953350): Enable this feature on Android as well after
     // confirming that this feature doesn't regress anything.
-    base::FEATURE_DISABLED_BY_DEFAULT
+    base::FEATURE_ENABLED_BY_DEFAULT
 #else
     base::FEATURE_ENABLED_BY_DEFAULT
 #endif
@@ -2218,7 +2219,7 @@ BASE_FEATURE_PARAM(bool,
                    kBoostRenderProcessForLoadingPrioritizePrerendering,
                    &kBoostRenderProcessForLoading,
                    "prioritize_prerendering",
-                   true);
+                   false);
 
 // If true is specified, kBoostRenderProcessForLoading feature only prioritizes
 // the renderer process that is used for prerendering. This is a part of an
