diff --git a/ash/webui/sample_system_web_app_ui/BUILD.gn b/ash/webui/sample_system_web_app_ui/BUILD.gn
index ce44da67bb222..73dbfa0bcce67 100644
--- a/ash/webui/sample_system_web_app_ui/BUILD.gn
+++ b/ash/webui/sample_system_web_app_ui/BUILD.gn
@@ -1,12 +1,10 @@
-# Copyright 2019 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//tools/typescript/ts_library.gni")
 
 assert(is_chromeos, "Sample System Web App is ChromeOS only")
-assert(!is_official_build,
-       "Sample System Web App is only built for unofficial builds")
 
 static_library("sample_system_web_app_ui") {
   sources = [
diff --git a/ash/webui/sample_system_web_app_ui/mojom/BUILD.gn b/ash/webui/sample_system_web_app_ui/mojom/BUILD.gn
index 79feb42f93d5e..6d5dd13c1f204 100644
--- a/ash/webui/sample_system_web_app_ui/mojom/BUILD.gn
+++ b/ash/webui/sample_system_web_app_ui/mojom/BUILD.gn
@@ -1,12 +1,10 @@
-# Copyright 2021 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
 assert(is_chromeos, "Sample System Web App is ChromeOS only")
-assert(!is_official_build,
-       "Sample System Web App is only built for unofficial builds")
 
 mojom("mojo_bindings") {
   sources = [
diff --git a/ash/webui/sample_system_web_app_ui/resources/trusted/BUILD.gn b/ash/webui/sample_system_web_app_ui/resources/trusted/BUILD.gn
index b262da1a8b7ba..5655202e77256 100644
--- a/ash/webui/sample_system_web_app_ui/resources/trusted/BUILD.gn
+++ b/ash/webui/sample_system_web_app_ui/resources/trusted/BUILD.gn
@@ -1,4 +1,4 @@
-# Copyright 2021 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
@@ -8,8 +8,6 @@ import("//ui/webui/resources/tools/build_webui.gni")
 import("//ui/webui/resources/tools/generate_grd.gni")
 
 assert(is_chromeos, "Sample System Web App is ChromeOS only")
-assert(!is_official_build,
-       "Sample System Web App is only built for unofficial builds")
 
 build_webui("build") {
   grd_prefix = "ash_sample_system_web_app"
diff --git a/ash/webui/sample_system_web_app_ui/resources/untrusted/BUILD.gn b/ash/webui/sample_system_web_app_ui/resources/untrusted/BUILD.gn
index 6cc7f87afb675..b9d1e70eb33d2 100644
--- a/ash/webui/sample_system_web_app_ui/resources/untrusted/BUILD.gn
+++ b/ash/webui/sample_system_web_app_ui/resources/untrusted/BUILD.gn
@@ -1,12 +1,10 @@
-# Copyright 2021 The Chromium Authors
+# Copyright 2026 The Chromium Authors and Alex313031
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
 import("//ui/webui/resources/tools/build_webui.gni")
 
 assert(is_chromeos, "Sample System Web App is ChromeOS only")
-assert(!is_official_build,
-       "Sample System Web App is only built for unofficial builds")
 
 build_webui("build") {
   grd_prefix = "ash_sample_system_web_app_untrusted"
diff --git a/ash/webui/sample_system_web_app_ui/sample_system_web_app_ui.h b/ash/webui/sample_system_web_app_ui/sample_system_web_app_ui.h
index e93621924aefd..7327b4b7a665c 100644
--- a/ash/webui/sample_system_web_app_ui/sample_system_web_app_ui.h
+++ b/ash/webui/sample_system_web_app_ui/sample_system_web_app_ui.h
@@ -5,9 +5,7 @@
 #ifndef ASH_WEBUI_SAMPLE_SYSTEM_WEB_APP_UI_SAMPLE_SYSTEM_WEB_APP_UI_H_
 #define ASH_WEBUI_SAMPLE_SYSTEM_WEB_APP_UI_SAMPLE_SYSTEM_WEB_APP_UI_H_
 
-#if defined(OFFICIAL_BUILD)
-#error Sample System Web App should only be included in unofficial builds.
-#endif
+
 
 #include <memory>
 
diff --git a/ash/webui/sample_system_web_app_ui/sample_system_web_app_untrusted_ui.h b/ash/webui/sample_system_web_app_ui/sample_system_web_app_untrusted_ui.h
index 145ef702d8d25..241c3227f1ab3 100644
--- a/ash/webui/sample_system_web_app_ui/sample_system_web_app_untrusted_ui.h
+++ b/ash/webui/sample_system_web_app_ui/sample_system_web_app_untrusted_ui.h
@@ -12,9 +12,6 @@
 #include "mojo/public/cpp/bindings/receiver.h"
 #include "ui/webui/untrusted_web_ui_controller.h"
 
-#if defined(OFFICIAL_BUILD)
-#error Sample System Web App should only be included in unofficial builds.
-#endif
 
 namespace content {
 class WebUI;
diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index bd63e688dbb96..9d22f1a5965aa 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -714,12 +714,10 @@ if (is_chromeos) {
       "//content/public/browser",
     ]
 
-    if (!is_official_build) {
-      deps += [
-        "//ash/webui/sample_system_web_app_ui",
-        "//ash/webui/sample_system_web_app_ui/mojom:mojo_bindings",
-      ]
-    }
+    deps += [
+      "//ash/webui/sample_system_web_app_ui",
+      "//ash/webui/sample_system_web_app_ui/mojom:mojo_bindings",
+    ]
   }
 
   source_set("platform_util_ash") {
diff --git a/chrome/browser/ash/system_web_apps/apps/BUILD.gn b/chrome/browser/ash/system_web_apps/apps/BUILD.gn
index c86dc6d2f2c5d..c3cdb2b91f37f 100644
--- a/chrome/browser/ash/system_web_apps/apps/BUILD.gn
+++ b/chrome/browser/ash/system_web_apps/apps/BUILD.gn
@@ -186,17 +186,15 @@ static_library("apps") {
     "//url",
   ]
 
-  if (!is_official_build) {
-    sources += [
-      "sample_system_web_app_info.cc",
-      "sample_system_web_app_info.h",
-    ]
+  sources += [
+    "sample_system_web_app_info.cc",
+    "sample_system_web_app_info.h",
+  ]
 
-    deps += [
-      "//ash/webui/sample_system_web_app_ui",
-      "//ash/webui/sample_system_web_app_ui/resources/trusted:resources",
-    ]
-  }
+  deps += [
+    "//ash/webui/sample_system_web_app_ui",
+    "//ash/webui/sample_system_web_app_ui/resources/trusted:resources",
+  ]
 }
 
 source_set("browser_tests") {
diff --git a/chrome/browser/ash/system_web_apps/apps/sample_system_web_app_info.h b/chrome/browser/ash/system_web_apps/apps/sample_system_web_app_info.h
index c80f8c0a8995c..d30b7ca6de16f 100644
--- a/chrome/browser/ash/system_web_apps/apps/sample_system_web_app_info.h
+++ b/chrome/browser/ash/system_web_apps/apps/sample_system_web_app_info.h
@@ -12,9 +12,6 @@ namespace web_app {
 struct WebAppInstallInfo;
 }  // namespace web_app
 
-#if defined(OFFICIAL_BUILD)
-#error Demo Mode should only be included in unofficial builds.
-#endif
 
 class SampleSystemAppDelegate : public ash::SystemWebAppDelegate {
  public:
diff --git a/chrome/browser/ash/system_web_apps/system_web_app_manager.cc b/chrome/browser/ash/system_web_apps/system_web_app_manager.cc
index a05f8507ea252..6f3728044c9d1 100644
--- a/chrome/browser/ash/system_web_apps/system_web_app_manager.cc
+++ b/chrome/browser/ash/system_web_apps/system_web_app_manager.cc
@@ -94,9 +94,7 @@
 #include "ui/base/l10n/l10n_util.h"
 #include "url/gurl.h"
 #include "url/origin.h"
-#if !defined(OFFICIAL_BUILD)
 #include "chrome/browser/ash/system_web_apps/apps/sample_system_web_app_info.h"
-#endif  // !defined(OFFICIAL_BUILD)
 
 namespace ash {
 
@@ -148,9 +146,7 @@ SystemWebAppDelegateMap CreateSystemWebApps(Profile* profile) {
         std::make_unique<graduation::GraduationAppDelegate>(profile));
   }
 
-#if !defined(OFFICIAL_BUILD)
   info_vec.push_back(std::make_unique<SampleSystemAppDelegate>(profile));
-#endif  // !defined(OFFICIAL_BUILD)
 
   SystemWebAppDelegateMap delegate_map;
   for (auto& info : info_vec) {
diff --git a/chrome/browser/chrome_browser_interface_binders_webui_parts_chromeos.cc b/chrome/browser/chrome_browser_interface_binders_webui_parts_chromeos.cc
index 5ceb0a65d293a..63308fcff49fe 100644
--- a/chrome/browser/chrome_browser_interface_binders_webui_parts_chromeos.cc
+++ b/chrome/browser/chrome_browser_interface_binders_webui_parts_chromeos.cc
@@ -168,10 +168,10 @@
 #include "media/capture/video/chromeos/mojom/camera_app.mojom.h"
 #include "mojo/public/cpp/bindings/binder_map.h"
 
-#if !defined(OFFICIAL_BUILD)
 #include "ash/webui/sample_system_web_app_ui/mojom/sample_system_web_app_ui.mojom.h"
 #include "ash/webui/sample_system_web_app_ui/sample_system_web_app_ui.h"
 #include "ash/webui/sample_system_web_app_ui/sample_system_web_app_untrusted_ui.h"
+#if !defined(OFFICIAL_BUILD)
 #include "ash/webui/status_area_internals/mojom/status_area_internals.mojom.h"
 #include "ash/webui/status_area_internals/status_area_internals_ui.h"
 #endif  // defined(OFFICIAL_BUILD)
@@ -587,10 +587,10 @@ void PopulateChromeWebUIFrameInterfaceBrokersTrustedPartsCros(
   registry.ForWebUI<ash::smb_dialog::SmbCredentialsDialogUI>();
   registry.ForWebUI<ash::MallUI>().Add<ash::mall::mojom::PageHandler>();
 
-#if !defined(OFFICIAL_BUILD)
   registry.ForWebUI<ash::SampleSystemWebAppUI>()
       .Add<ash::mojom::sample_swa::PageHandlerFactory>();
 
+#if !defined(OFFICIAL_BUILD)
   registry.ForWebUI<ash::StatusAreaInternalsUI>()
       .Add<ash::mojom::status_area_internals::PageHandler>();
 #endif  // !defined(OFFICIAL_BUILD)
@@ -630,10 +630,8 @@ void PopulateChromeWebUIFrameInterfaceBrokersUntrustedPartsCros(
   registry.ForWebUI<ash::ScannerFeedbackUntrustedUI>()
       .Add<ash::mojom::scanner_feedback_ui::PageHandler>();
 
-#if !defined(OFFICIAL_BUILD)
   registry.ForWebUI<ash::SampleSystemWebAppUntrustedUI>()
       .Add<ash::mojom::sample_swa::UntrustedPageInterfacesFactory>();
-#endif  // !defined(OFFICIAL_BUILD)
 }
 
 }  // namespace chrome::internal
diff --git a/chrome/browser/ui/webui/ash/config/BUILD.gn b/chrome/browser/ui/webui/ash/config/BUILD.gn
index eafab5f60c6d5..8f8ca4d98e3d3 100644
--- a/chrome/browser/ui/webui/ash/config/BUILD.gn
+++ b/chrome/browser/ui/webui/ash/config/BUILD.gn
@@ -119,9 +119,7 @@ static_library("config") {
     "//ui/webui",
   ]
 
-  if (!is_official_build) {
-    deps += [ "//ash/webui/sample_system_web_app_ui" ]
-  }
+  deps += [ "//ash/webui/sample_system_web_app_ui" ]
 }
 
 source_set("unit_tests") {
diff --git a/chrome/browser/ui/webui/ash/config/ash_web_ui_config_manager.cc b/chrome/browser/ui/webui/ash/config/ash_web_ui_config_manager.cc
index 8071c4efb25dd..7f8354eb2e70c 100644
--- a/chrome/browser/ui/webui/ash/config/ash_web_ui_config_manager.cc
+++ b/chrome/browser/ui/webui/ash/config/ash_web_ui_config_manager.cc
@@ -148,10 +148,8 @@
 #include "url/gurl.h"
 #include "url/url_constants.h"
 
-#if !defined(OFFICIAL_BUILD)
 #include "ash/webui/sample_system_web_app_ui/sample_system_web_app_ui.h"
 #include "ash/webui/sample_system_web_app_ui/sample_system_web_app_untrusted_ui.h"
-#endif  // !defined(OFFICIAL_BUILD)
 
 namespace content {
 class WebUI;
@@ -482,8 +480,8 @@ void AshWebUIConfigManager::RegisterWebUIConfigs() {
       base::BindRepeating(vc_background_ui::CreateVcBackgroundUI)));
   AddWebUIConfig(std::make_unique<GrowthInternalsUIConfig>());
   AddWebUIConfig(std::make_unique<FloatingWorkspaceUIConfig>());
-#if !defined(OFFICIAL_BUILD)
   AddWebUIConfig(std::make_unique<SampleSystemWebAppUIConfig>());
+#if !defined(OFFICIAL_BUILD)
   AddWebUIConfig(std::make_unique<StatusAreaInternalsUIConfig>());
 #endif  // !defined(OFFICIAL_BUILD)
 }
@@ -518,10 +516,8 @@ void AshWebUIConfigManager::RegisterUntrustedWebUIConfigs() {
   AddUntrustedWebUIConfig(std::make_unique<MakoUntrustedUIConfig>());
   AddUntrustedWebUIConfig(std::make_unique<FocusModeUntrustedUIConfig>());
   AddUntrustedWebUIConfig(std::make_unique<ScannerFeedbackUntrustedUIConfig>());
-#if !defined(OFFICIAL_BUILD)
   AddUntrustedWebUIConfig(
       std::make_unique<SampleSystemWebAppUntrustedUIConfig>());
-#endif  // !defined(OFFICIAL_BUILD)
 }
 
 void AshWebUIConfigManager::AddWebUIConfig(
