diff --git a/chrome/browser/ui/views/toolbar/avatar_toolbar_button_interface.cc b/chrome/browser/ui/views/toolbar/avatar_toolbar_button_interface.cc
index 7f3e55f684fb7..983f380a7e6ca 100644
--- a/chrome/browser/ui/views/toolbar/avatar_toolbar_button_interface.cc
+++ b/chrome/browser/ui/views/toolbar/avatar_toolbar_button_interface.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/views/toolbar/avatar_toolbar_button_interface.h"
 
+#include "base/command_line.h"
 #include "base/time/time.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/browser_element_identifiers.h"
@@ -12,6 +13,19 @@
 
 // static
 bool AvatarToolbarButtonInterface::CanShowForProfile(Profile* profile) {
+  const auto avatar_button_visibility =
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          "show-avatar-button");
+  if (avatar_button_visibility == "always") {
+    return true;
+  }
+  if (avatar_button_visibility == "incognito-and-guest") {
+    return profile->IsIncognitoProfile() || profile->IsGuestSession();
+  }
+  if (avatar_button_visibility == "never") {
+    return false;
+  }
+
 #if BUILDFLAG(IS_CHROMEOS)
   // ChromeOS only badges Incognito, Guest, and captive portal signin icons in
   // the browser window.
