diff --git a/BUILD.gn b/BUILD.gn
index 76391be1666..95f05d532c1 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1819,10 +1819,6 @@ config("toolchain") {
 
   if (v8_current_cpu == "x86") {
     defines += [ "V8_TARGET_ARCH_IA32" ]
-    if (is_win) {
-      # Ensure no surprising artifacts from 80bit double math with x86.
-      cflags += [ "/arch:SSE2" ]
-    }
   }
   if (v8_current_cpu == "x64") {
     defines += [ "V8_TARGET_ARCH_X64" ]
