diff --git a/BUILD.gn b/BUILD.gn
index c6132807ff6..3c543f8e0cf 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1839,10 +1839,6 @@ config("toolchain") {
 
   if (v8_current_cpu == "x86") {
     defines += [ "V8_TARGET_ARCH_IA32" ]
-    if (is_win) {
-      # Ensure no surprising artifacts from 80bit double math with x86.
-      cflags += [ "/arch:SSE2" ]
-    }
   }
   if (v8_current_cpu == "x64") {
     defines += [ "V8_TARGET_ARCH_X64" ]
