diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index 973a46b2a34bb..89d9c6ccd808d 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1022,7 +1022,7 @@ if (is_win) {
   if (bundle_widevine_cdm) {
     bundle_data("widevine_cdm_library_binaries") {
       sources = [ "$root_out_dir/$widevine_cdm_path/libwidevinecdm.dylib" ]
-      if (enable_widevine_cdm_host_verification) {
+      if (bundle_widevine_cdm_signature) {
         sources +=
             [ "$root_out_dir/$widevine_cdm_path/libwidevinecdm.dylib.sig" ]
       }
